db/dcrpg/pgblockchain.go

```diff
@@ -711,11 +712,15 @@ func (pgb *ChainDB) HeightHashDB() (uint64, string, error) {
 
 // Height uses the last stored height.
 func (pgb *ChainDB) Height() uint64 {
+	pgb.RLock()
+	defer pgb.RUnlock()
 	return uint64(pgb.bestBlock)
 }
 
 // HashStr uses the last stored block hash.
 func (pgb *ChainDB) HashStr() string {
+	pgb.RLock()
+	defer pgb.RUnlock()
```

Comment:
These are warranted, for sure, but please do audit for calls to these to make sure we won't introduce a deadlock somewhere.

Reply:
Good call. Doesn't look like any chance of deadlocks, but there were a couple more reads that needed protection. 

The mutex is only used on the `bestBlock` and `bestBlockHash` members of `ChainDB`, so rather than embed the mutex in `ChainDB`, would it be better to have it as a named member?

Reply:
That's a good idea, although it might be cleaner to avoid a named mutex in this case.  I might make it similar to the changes in https://github.com/decred/dcrdata/pull/756.  It's a similar situation here (big struct where it's unclear what fields the mutex is supposed to guard).  i.e. something like:

```go
type BestBlock struct {
	sync.RWMutex
	Height int64
	Hash string
}
type ChainDB struct {
	...
	bestBlock          BestBlock
	...
}
// Height uses the last stored height.
func (pgb *ChainDB) Height() uint64 {
	pgb.bestBlock.RLock()
	defer pgb.bestBlock.RUnlock()
	return uint64(pgb.bestBlock.Height)
}
...
```